argus/backends/azure/client.py

```diff
@@ -0,0 +1,240 @@
+#    under the License.
+import abc
+import random
+import six
+
+from argus import util
+from azure.common.credentials import UserPassCredentials
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.storage import StorageManagementClient
+from azure.mgmt.network import NetworkManagementClient
+from azure.mgmt.compute import ComputeManagementClient
+from azure.storage import CloudStorageAccount
+
+VM_REFERENCE = {
+    'windows': {
+        'publisher': 'MicrosoftWindowsServerEssentials',
+        'offer': 'WindowsServerEssentials',
+        'sku': 'WindowsServerEssentials',
+        'version': 'latest'
+    }
+}
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseAzureClient(object):
+    """Class for managing Azure instances
+
+    """
+    def __init__(self, subscription_id=None, username=None, password=None,
+                 resource_group_name=None, storage_account_name=None,
+                 availability_zone=None, image_vhd_path=None, vm_username=None,
+                 vm_password=None):
+        self._subscription_id = subscription_id
+        self._username = username
+        self._password = password
+        self._resource_group_name = resource_group_name
+        self._storage_account_name = storage_account_name
+        self._availability_zone = availability_zone
+        self._image_vhd_path = image_vhd_path
+        self._vm_username = vm_username
+        self._vm_password = vm_password
+        self._credentials = UserPassCredentials(username, password)
+        self._resource_client = ResourceManagementClient(self._credentials, subscription_id)
+        self._storage_client = StorageManagementClient(self._credentials, subscription_id)
+        self._compute_client = ComputeManagementClient(self._credentials, subscription_id)
+        self._network_client = NetworkManagementClient(self._credentials, subscription_id)
+
+    @abc.abstractmethod
+    def create_vm(self):
+        pass
+
+    @abc.abstractmethod
+    def destroy_vm(self):
+        pass
+
+    @abc.abstractmethod
+    def create_resource_group(self):
+        pass
+
+    @abc.abstractmethod
+    def create_storage_account(self):
+        pass
+
+    @abc.abstractmethod
+    def create_nic(self):
+        pass
+
+    @abc.abstractmethod
+    def get_floating_ip(self):
+        pass
+
+
+class AzureClient(BaseAzureClient):
+    """Class for managing Azure instances
+
+    """
+    def __init__(self, subscription_id=None, username=None, password=None,
+                 resource_group_name=None, storage_account_name=None,
+                 availability_zone=None, image_vhd_path=None, vm_username=None,
+                 vm_password=None):
+        super(AzureClient, self).__init__(subscription_id, username,
+            password, resource_group_name, storage_account_name, availability_zone,
+            image_vhd_path, vm_username, vm_password)
+
+        self._vm_name = "argusvm-" + str(random.randint(0,10000))
```

Comment:
This is not really that random, it would be better to use a random UUID seed or simply take the current time as a suffix, as there cannot be 2 dates the same.